api/billingd/service/service.go

```diff
@@ -755,6 +771,59 @@ func (s *Service) DeleteCustomer(ctx context.Context, req *pb.DeleteCustomerRequ
 	return &pb.DeleteCustomerResponse{}, nil
 }
 
+func (s *Service) GetCustomerUsage(
+	ctx context.Context,
+	req *pb.GetCustomerUsageRequest,
+) (*pb.GetCustomerUsageResponse, error) {
+	doc, err := s.getCustomer(ctx, "_id", req.Key)
+	if err != nil {
+		return nil, err
+	}
+	usage := make(map[string]*pb.Usage)
+	for k, u := range doc.DailyUsage {
+		if product, ok := s.products[k]; ok {
```

Comment:
Q: Does a `log.Warnf` for the `!ok` case a good idea? Like, some daily usage report for some unknown product.

Reply:
Sure, can add

Reply:
Actually, don't think it makes sense here, but does when the usage is added, since the caller controls the product key.

Reply:
Eh, I take that back... thinking too fast this morning. Will add here.